languages/elixir/concepts/tail-call-recursion/introduction.md

```diff
@@ -1,9 +1,13 @@
-When recursing through enumerables [lists, bitstrings, strings], there are often two concerns:
+When recursing through enumerables (lists, bitstrings, strings), there are often two concerns:
 
 - how much memory is required to store the trail of recursive function calls
 - how to build the solution efficiently
 
-To deal with these concerns an _accumulator_ may be used to pass the state of the function _so far_ until the result reaches the _base case_ and returns the value. An accumulator is a variable that is passed along in addition to the data.
+To deal with these concerns an _accumulator_ may be used.
+
+An accumulator is a variable that is passed along in addition to the data. It is used to pass the current state of the function's execution, from function call to function call, until the _base case_ is reached. In the base case, the accumulator is used to return the final value of the recursive function call.
+
+Accumulators should be initialized by the function's author, not the function's user. To achieve this, declare two functions - a public function that takes just the necessary data as arguments and initializes the accumulator, and a private function that also takes an accumulator. In Elixir, it is a common pattern to prefix the private function's name with `do_`.
```

Comment:
I think accumulators are a very tricky topic for people new to functional programming. I wanted to explain them more explicitly. I would love a pair of eyes on this text that aren't doing functional programming very often. I will make assumptions and nominate @iHiD - does this explanation of accumulators make sense to you?